### DISCUSSION TITLE
Flatpak release for Studio

### DISCUSSION THREAD

**Discussion by rgov** [foxglove/community#815] [Q&A]
SteamOS on the Steam Deck requires that packages are installed with [Flatpak](https://flatpak.org). Users have [hacked on getting the snap running on SteamOS](https://discourse.ros.org/t/for-linux-users-install-the-foxglove-studio-snap/26451) but it's unsupported.

I made another hacky attempt at [converting the snap to a Flatpak](https://gist.github.com/rgov/9f3b4ee00d84b1878a70b07803166ad2) (**if you just found this thread, please note this manifest almost certainly doesn't work**) but an officially supported one would be terrific.

**Comment by bradenwagstaff** [comment]
I have little experience with Flatpack, but I ran this on my steam deck and it can't seem to find a file it needs from squashfs-tools

```
(1)(deck@steamdeck Foxglove)$ flatpak-builder --force-clean /home/deck/Foxglove/build dev.foxglove.Studio.yaml
Downloading sources
Failed to download sources: module squashfs: Can't find file at xz_support.patch
(1)(deck@steamdeck Foxglove)$ 
```

**Reply by rgov** [reply]
The top of the file includes a list of commands to run first. Sorry I didn't mention that.

**Reply by bradenwagstaff** [reply]
Even when I make sure those are installed, it still can't find that path. I think it's this line it has a problem with 
https://gist.github.com/rgov/9f3b4ee00d84b1878a70b07803166ad2#file-dev-foxglove-studio-yaml-L31

**Comment by bradenwagstaff** [comment]
Tested this on my steam deck and it worked. Be sure to install the packages in the top of the .yaml file and it should work

![image](https://github.com/foxglove/community/assets/46809823/d66028b7-1234-49ab-a647-c9dc748182d2)

**Comment by achim-k** [comment]
[electron-builder](https://www.electron.build/configuration/flatpak) supports creating single-file bundle flatpaks out of the box. Just a [minor tweak](https://github.com/foxglove/studio/commit/39fac6518803cf64d175a7a0dd94ee7aa6271f80) was necessary to be able to build the Studio flatpak:


### Install build deps
```sh
sudo apt install flatpak flatpak-builder
flatpak install flathub org.freedesktop.Platform//22.08
flatpak install flathub org.freedesktop.Sdk//22.08
flatpak install org.electronjs.Electron2.BaseApp//22.08
```

### Clone Studio and build flatpak

```sh
git clone git@github.com:foxglove/studio.git
cd studio
git merge origin/achim/flatpak_fixes  # includes icon size tweak
yarn
yarn desktop:build:prod
DEBUG="@malept/flatpak-bundler" yarn package --linux flatpak
```

### Install flatpak
```sh
flatpak install dist/foxglove-studio-*.flatpak
flatpak run dev.foxglove.studio
```

**Comment by AlexKlimaj** [comment]
Hello, is there an official flatpack for the steam deck? I am still running the one from the USB drive at ROSCon.

**Comment by tsdaemon** [comment]
The problem still stands, and builing flatpak from source is not available anymore...
